k8s.gcr.io/lib.sh

```diff
@@ -0,0 +1,279 @@
+    gsutil iam ch \
+        "group:${GCR_ADMINS}:objectAdmin" \
+        $(gcs_bucket_for "${project}" "${region}")
+    gsutil iam ch \
+        "group:${GCR_ADMINS}:legacyBucketOwner" \
+        $(gcs_bucket_for "${project}" "${region}")
+}
+
+# Grant write privileges to a group
+# $1: The GCP project
+# $2: The googlegroups group
+function empower_group() {
+    if [ $# -lt 2 -o $# -gt 3 -o -z "$1" -o -z "$2" ]; then
+        echo "empower_group(project, group_name, [region]) requires 2 or 3 arguments" > /dev/stderr
+        return 1
+    fi
+    project="$1"
+    group="$2"
+    region="${3:-}"
+
+    gsutil iam ch \
+        "group:${group}:objectAdmin" \
+        $(gcs_bucket_for "${project}" "${region}")
+    gsutil iam ch \
+        "group:${group}:legacyBucketReader" \
+        $(gcs_bucket_for "${project}" "${region}")
+}
+
+# Grant full privileges to the GCR promoter bot
+# $1: The GCP project
+function empower_promoter() {
+    if [ $# -lt 1 -o $# -gt 2 -o -z "$1" ]; then
+        echo "empower_promoter(project, [region]) requires 1 or 2 arguments" > /dev/stderr
+        return 1
+    fi
+    project="$1"
+    region="${2:-}"
+
+    acct=$(svc_acct_for "${project}" "${PROMOTER_SVCACCT}")
+
+    if ! gcloud --project "${project}" iam service-accounts describe "${acct}" >/dev/null 2>&1; then
+        gcloud --project "${project}" \
+            iam service-accounts create \
+            "${name}" \
+            --display-name="k8s-infra container image promoter"
+    fi
+
+    # Grant admins access to do admin stuff.
+    gsutil iam ch \
+        "serviceAccount:${acct}:objectAdmin" \
+        $(gcs_bucket_for "${project}" "${region}")
+    gsutil iam ch \
+        "serviceAccount:${acct}:legacyBucketOwner" \
+        $(gcs_bucket_for "${project}" "${region}")
+}
+
+# Configure bucket lifecycle for a prod repo
+# $1: The GCP project
+function ensure_prod_lifecycle() {
+    if [ $# -lt 1 -o $# -gt 2 -o -z "$1" ]; then
+        echo "ensure_prod_lifecycle(project, [region]) requires 1 or 2 arguments" > /dev/stderr
+        return 1
+    fi
+    project="$1"
+    region="${2:-}"
+
+    # Set lifecycle policies.
+    # TODO: do we want this?  It will inhibit promoter's GC, but will protect
+    # against mistakenly nuking images we wanted to keep.
+    #TODO: if we keep this, maybe we want to lock this (can't be overridden)?
+    gsutil retention set 5y $(gcs_bucket_for "${project}" "${region}")
+}
+
+# Configure bucket lifecycle for a staging repo
+# $1: The GCP project
+function ensure_staging_lifecycle() {
+    if [ $# -lt 1 -o $# -gt 2 -o -z "$1" ]; then
+        echo "ensure_staging_lifecycle(project, [region]) requires 1 or 2 arguments" > /dev/stderr
+        return 1
+    fi
+    project="$1"
+    region="${2:-}"
+
+    # Set lifecycle policies.
+    gsutil retention set 30d $(gcs_bucket_for "${project}" "${region}")
+    echo '
+        {
+          "rule": [
+            {
+              "condition": {
+                "age": 30
```

Comment:
If the `30` here is written to match the `30d` in line 255, I would throw this into a function and reuse the number. Also might as well do a heredoc instead to pass in variables to the multiline string.

Reply:
They are not necessarily the same value

Reply:
Ack